### ISSUE TITLE
Function declaration between const and another function makes var read-only

### ISSUE THREAD

**Issue by pornel** [babel/babel#1199]
I think `bug = 2` is valid, because there's `var bug` in that function. babel 5.0.10 reports an error.

```js
'use strict';

const bug = 1;

function foo() {}

function bar() {
	var bug;
	bug = 2;
}
```

> SyntaxError: test.js: Line 9: "bug" is read-only
>   7 | function bar() {
>   8 | 	var bug;
> \>  9 | 	bug = 2;
>     |  ^
>  10 | }

However, this parses fine:

```js
'use strict';

const bug = 1;

// function foo() {}

function bar() {
	var bug;
	bug = 2;
}
```

**Comment by sebmck** [comment]
Something wacky is going on with functions that I need to fix ASAP. I believe this is the exact same cause of #1169.

**Comment by sebmck** [comment]
Fixed via de5520a94f9dcfd602c1c444da6f71ff06dd7791

**Comment by sebmck** [comment]
Fixed as of 5.0.11, thanks!
